### PR TITLE
Fix nested Salt Providers not inheriting a parent provider's `enableStyleInjection` value

### PR DESCRIPTION
**Pull request by DavieReid**
Root `StyleInjectionProvider` overrides nested `StyleInjectionProvider`

### CODE REVIEW

File: packages/core/src/salt-provider/SaltProvider.tsx [hunk 1]
```diff
@@ -209,7 +209,7 @@ function InternalSaltProvider({
 }
 
 export function SaltProvider({
-  enableStyleInjection = true,
+  enableStyleInjection,
```
Comment (DavieReid) [inline packages/core/src/salt-provider/SaltProvider.tsx:212]: default comes from context now

File: packages/core/stories/style-injection/style-injection.disabled.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export default {
+  title: "Core/Style Injection",
+  component: SaltProvider,
+};
+
+export const RootDisabled: StoryFn = () => (
+  <SaltProvider enableStyleInjection={false}>
+    <Button variant="primary">Root - style injection disabled</Button>
+    <SaltProvider>
+      <Button variant="primary">Nested - style injection enabled</Button>
```
Comment (ZarkoPernar) [inline packages/core/stories/style-injection/style-injection.disabled.stories.tsx:14]: disabled?
```suggestion
      <Button variant="primary">Nested - style injection disabled</Button>
```

Comment (DavieReid) [reply packages/core/stories/style-injection/style-injection.disabled.stories.tsx:14]: No, it's nested inside a provider that has style injection enabled (line 13).

This test should show 2 buttons with no styles as the root style provider trumps the nested style provider

Comment (DavieReid) [reply packages/core/stories/style-injection/style-injection.disabled.stories.tsx]: Updated the text to make clarify the expectations

**Review by joshwooding — CHANGES_REQUESTED** [review]
Can you add a changeset please :) Other than that looks good

**Review by joshwooding — APPROVED** [review]

**Pull request merged**